app/views/site/view_gene_expression_heatmap.html.erb

```diff
@@ -0,0 +1,90 @@
+    <% end %>
+    <% if @allow_firecloud_access && @user_can_compute %>
+      <% if @allow_computes %>
+        <li role="presentation" class="study-nav" id="study-analysis-nav"><a href="#study-analysis" data-toggle="tab">Analysis <i class="fas fa-fw fa-tasks"></i></a></li>
+      <% else %>
+        <li role="presentation" class="study-nav disabled" id="study-analysis-nav"><a href="#study-analysis" data-toggle="tooltip" title="Study workspaces are currently unavailable - please try again later.">Analysis <i class="fas fa-fw fa-tasks"></i></a></li>
+  <% end %>
+    <% end %>
+    <% if @allow_firecloud_access && @user_can_edit %>
+      <% if @allow_edits %>
+        <li role="presentation" class="study-nav" id="study-settings-nav"><a href="#study-settings" data-toggle="tab">Settings <i class="fas fa-fw fa-cogs"></i></a></li>
+      <% else %>
+        <li role="presentation" class="study-nav disabled" id="study-settings-nav"><a href="#study-settings" data-toggle="tooltip" title="Study workspaces are currently unavailable - please try again later.">Settings <i class="fas fa-fw fa-cogs"></i></a></li>
+      <% end %>
+    <% end %>
+  </ul>
+  <div class="tab-content top-pad">
+    <div class="tab-pane active in" id="study-summary" role="tabpanel">
+      <%= render partial: 'study_description_view' %>
+    </div>
+    <div class="tab-pane" id="study-visualize" role="tabpanel">
+      <% if @study.can_visualize? %>
+        <%= render partial: 'view_gene_expression_heatmap' %>
+      <% end %>
+    </div>
+    <% if @user_can_download %>
+      <div class="tab-pane" id="study-download" role="tabpanel">
+        <%= render partial: 'study_download_data' %>
+      </div>
+    <% end %>
+    <% if @allow_firecloud_access && @allow_computes && @user_can_compute %>
+      <div class="tab-pane" id="study-analysis" role="tabpanel">
+        <%= render partial: 'study_analysis' %>
+      </div>
+    <% end %>
+    <% if @allow_firecloud_access && @allow_edits && @user_can_edit %>
+      <div class="tab-pane" id="study-settings" role="tabpanel">
+        <div class="row">
+          <div class="col-xs-12" id="study-settings-form-target">
+            <%= render partial: 'study_settings_form' %>
+          </div>
+        </div>
+      </div>
+    <% end %>
+  </div>
+</div>
+
+<div class="modal fade" id="update-study-settings-modal" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h4 class="text-center">Updating Study... Please wait<br/></h4>
+      </div>
+      <div class="modal-body">
+        <div class="spinner-target" id="update-study-settings-spinner"></div>
+      </div>
+      <div class="modal-footer">
+        <button class="close" data-dismiss="modal">×</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
```

Comment:
I won't remember this code needs to be changed in two places, if that's ever needed.  

De-duplicating this code and [its other copy](https://github.com/broadinstitute/single_cell_portal_core/pull/250/files#diff-44ceb12e4c5a5d3a76b268ad3ab0c872R82) added here would fix that lurking maintainability issue.  If a quicker solution is desired, a code comment noting as much would also mitigate this.

Reply:
Yes, you're correct.  Also, there's a much better way to write this function.  But I'll abstract it out and just make a function call so we're not duplicating code.